sbot/metadata.py

```diff
@@ -1,85 +1,111 @@
 """
 Implementation of loading metadata.
 
-Metadata is a dictionary of arbitrary information about the environment that the robot is
-running in. It usually includes the starting zone and a flag indicating whether we are in
+Metadata is a dictionary of information about the environment that the robot is running in.
+It usually includes the starting zone and a flag indicating whether we are in
 competition or development mode. Metadata is stored in a JSON file, typically on a
 competition USB stick. The environment variable SBOT_METADATA_PATH specifies a directory
-that is (recursively) searched for a JSON file to load.
+where it, and its children, are searched for the JSON file to load.
 
 Example metadata file:
-
-    {
-        "arena": "A",
-        "zone": 2,
-        "is_competition": true
-    }
+```json
+{
+    "zone": 2,
+    "is_competition": true
+}
+```
 """
+from __future__ import annotations
 
 import json
 import logging
 import os
-from typing import Any, Dict, Optional
+from pathlib import Path
+from typing import TypedDict
 
-LOGGER = logging.getLogger(__name__)
+from .exceptions import MetadataKeyError
 
-METADATA_ENV_VAR = "SBOT_METADATA_PATH"
+logger = logging.getLogger(__name__)
 
+# The name of the environment variable that specifies the path to search
+# for metadata USB sticks
+METADATA_ENV_VAR = "SBOT_METADATA_PATH"
+# The name of the metadata file
+METADATA_NAME = "metadata.json"
 
-class MetadataKeyError(KeyError):
-    """Raised when trying to access a metadata key for which no value exists."""
 
-    def __init__(self, key: str):
-        self.key = key
+class Metadata(TypedDict):
+    """
+    The structure of the metadata dictionary.
 
-    def __str__(self) -> str:
-        return f"Key {self.key!r} not present in metadata, or no metadata was available"
+    :param is_competition: Whether the robot is in competition mode
+    :param zone: The zone that the robot is in
+    """
+    is_competition: bool
+    zone: int
 
 
-class MetadataNotReadyError(RuntimeError):
-    """Raised when trying to access metadata before it has been loaded."""
+# The default metadata to use if no file is found
+DEFAULT_METADATA: Metadata = {
+    "is_competition": False,
+    "zone": 0,
+}
 
-    def __str__(self) -> str:
-        return (
-            "Metadata (e.g. zone or is_competition) can only be used after"
-            " wait_start has been called"
-        )
 
+def load() -> Metadata:
+    f"""
```

Comment:
Does this work with `help()`?

Reply:
No, guess I can't template that.

Reply:
05a96aae1e73a3fe1a7ff92f3360334b24f3095e